controllers/topics.js

```diff
@@ -11,320 +11,355 @@ const router = express.Router({ mergeParams: true });
+
+	api(req).patch(`/lessons/${req.params.topicId}`, {
+		json: data, // TODO: sanitize
+	}).then((_) => {
+		if (req.query.json) {
+			res.json(_);
+		} else {
+			// sends a GET request, not a PATCH
+			res.redirect(`/${context}/${req.params.courseId}/topics/${req.params.topicId
+			}${req.query.courseGroup ? `?courseGroup=${req.query.courseGroup}` : ''}`);
+		}
+	}).catch((_) => {
+		res.sendStatus(500);
+	});
 });
 
-router.delete('/:topicId', function (req, res, next) {
-    api(req).delete('/lessons/' + req.params.topicId).then(_ => {
-        res.sendStatus(200);
-    }).catch(err => {
-        next(err);
-    });
+router.delete('/:topicId', (req, res, next) => {
+	api(req).delete(`/lessons/${req.params.topicId}`).then((_) => {
+		res.sendStatus(200);
+	}).catch((err) => {
+		next(err);
+	});
 });
 
-router.delete('/:topicId/materials/:materialId', function (req, res, next) {
-    api(req).patch('/lessons/' + req.params.topicId, {
-        json: {
-            courseId: req.params.courseId,
-            $pull: {
-                materialIds: req.params.materialId
-            }
-        }
-    }).then(_ => {
-        api(req).delete('/materials/' + req.params.materialId).then(_ => {
-            res.sendStatus(200);
-        });
-    });
+router.delete('/:topicId/materials/:materialId', (req, res, next) => {
+	api(req).patch(`/lessons/${req.params.topicId}`, {
+		json: {
+			courseId: req.params.courseId,
+			$pull: {
+				materialIds: req.params.materialId,
+			},
+		},
+	}).then((_) => {
+		api(req).delete(`/materials/${req.params.materialId}`).then((_) => {
+			res.sendStatus(200);
+		});
+	});
 });
 
 router.get('/:topicId/edit', editTopicHandler);
 
 async function createNewNexBoards(req, res, contents = []) {
-    return await Promise.all(contents.map(async content => {
-        if (content.component === "neXboard" && content.content.board === '0') {
-            try {
-                const board = await getNexBoardAPI().createBoard(
-                    content.content.title,
-                    content.content.description,
-                    await getNexBoardProjectFromUser(req, res.locals.currentUser),
-                    'schulcloud');
-
-                content.content.title = board.title;
-                content.content.board = board.id;
-                content.content.url = board.publicLink;
-                content.content.description = board.description;
-
-                return content;
-
-            } catch (err) {
-                logger.error(err);
-
-                return undefined;
-            }
-        } else
-            return content;
-    }));
+	return await Promise.all(contents.map(async (content) => {
+		if (content.component === 'neXboard' && content.content.board === '0') {
+			try {
+				const board = await getNexBoardAPI().createBoard(
+					content.content.title,
+					content.content.description,
+					await getNexBoardProjectFromUser(req, res.locals.currentUser),
```

Comment:
execute await getNexBoardProjectFromUser(req, res.locals.currentUser) before and add the response in createBoard
for more readable code